src/algos/topologies/collections.py

```diff
@@ -55,6 +55,32 @@ def __init__(self, config: ConfigType, rank: int):
     def generate_graph(self) -> None:
         self.graph = nx.circular_ladder_graph(self.num_users) # type: ignore
 
+class TreeTopology(BaseTopology):
+    def __init__(self, config: ConfigType, rank: int, children: int = 2):
+        super().__init__(config, rank)
+        self.children = children
+
+    def generate_graph(self) -> None:
```

Comment:
Your generate graph looks correct but I wonder if networkX has already an implementation that we can use off-the-shelf

Reply:
Ok I added this code because I originally could only find balanced_tree as a way to generate a tree graph. Which would take in a specific number for children and height and this was messy to implement (as the height was messy to find and nodes would still need to be added/removed for certain node numbers)

I took another look and found full_rary_tree (which IDK how I missed this originally). I will work on implementing this. My fault should have been more lazy.